src/applications/inter-app-communication.md

```diff
@@ -0,0 +1,46 @@
+## Inter-app Communication
+
+### DNS Service Discovery
+
+Deis Workflow supports deploying a single app composed of a system of processes. Each Deis Workflow app communicates on a single port, so communicating with another Workflow app means finding that app's address and port. All Workflow apps are mapped to port 80 externally, so finding its IP address is the only challenge. Workflow creates a [Kubernetes Service](https://kubernetes.io/docs/user-guide/services/) for each app, which effectively assigns a name and one cluster-internal IP address to an app. The DNS service running in the cluster adds and removes DNS records which point from the app name to its IP address as services are added and removed. Deis Workflow apps, then, can simply send requests to the domain name given to the service, which is "app-name.app-namespace".
+
+For example, if you deployed a Deis Workflow app like this:
+
+    $ deis create http-server --no-remote
+    $ deis pull deis/example-dockerfile-python:latest -a http-server
+
+You can deploy a second app and communicate with the first one:
+
+    $ deis create http-client --no-remote
+    $ # Need to set a port because the docker image doesn't use one.
+    $ deis config:set PORT=8080 -a http-client
+    $ deis config:set HOSTNAME=http-server.http-server
+    $ deis pull -a http-client willfarrell/ping:latest
```

Comment:
Running `deis pull` on this image fails because it does not listen on port 8080 and fails the health check.

```
><> deis pull -a http-client willfarrell/ping:latest
Creating build... Error: Unknown Error (400): {"detail":"(app::deploy): Readiness probe failed: dial tcp 172.17.0.26:8080: getsockopt: connection refused"}
```